File: Begginers/C++/numero_primo_1165.cpp
#include <iostream>

int main() {
    int entrada;
    scanf("%d", &entrada);

    for (int j = 0; j < entrada; j++) {
        int N;
        scanf("%d", &N);
        bool ehPrimo = true;

        if (N <= 1) {
            ehPrimo = false;
        } else {
            for (int i = 2; i * i <= N; i++) {
                if (N % i == 0) {
                    ehPrimo = false;
                    break;
                }
            }
        }

        if (ehPrimo) {
            printf("%d eh primo\n", N);
        } else {
            printf("%d nao eh primo\n", N);
        }
    }

    return 0;
}

File: Begginers/Python/numero_primo_1165.py
entrada = int(input())

for j in range(entrada):
    N = int(input())
    lista = []
    
    for i in range(2, N + 1):
        if N % i == 0:
           lista.append(i)
           
    if len(lista) > 1:
       print(f"{N} nao eh primo")
    else:
       print(f"{N} eh primo")
